common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -241,10 +337,164 @@ public static ServerSocket getSocket(final TServerSocket thriftSocket) {
    * @return the InetSocketAddress of the master
    */
   public static InetSocketAddress getMasterAddress(TachyonConf conf) {
-    String masterHostname =
-        conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
+    String masterHost = conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
     int masterPort = conf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
-    return new InetSocketAddress(masterHostname, masterPort);
+    return new InetSocketAddress(masterHost, masterPort);
+  }
+
+  /**
+   * Helper method to get the {@link InetSocketAddress} connect address of the master.
+   *
+   * @param conf the configuration of Tachyon
+   * @return the master's connect address
+   */
+  public static InetSocketAddress getMasterConnectAddress(TachyonConf conf) {
+    if (conf.get(Constants.MASTER_BIND_HOST, "localhost").equals("0.0.0.0")) {
```

Comment:
I'm a bit confused by this logic.

1. Why set the default as "localhost", if that's going to necessarily compare false with "0.0.0.0". Wouldn't it be faster to set it to an empty string?
2. What does a setting of "0.0.0.0" signify anyways? If it's meant to signal that Tachyon should figure out the host name, why not just do that if the property is unset?